core/field_variable.js

```diff
@@ -177,10 +164,14 @@ Blockly.FieldVariable.dropdownCreate = function() {
  */
 Blockly.FieldVariable.prototype.onItemSelected = function(menu, menuItem) {
   var itemText = menuItem.getValue();
```

Comment:
Agreed. I'm setting it to id for now.
In the next CL I will change field_variable's setValue() to take in an id since all items are now ids. Then we can get rid of itemText.